server/app/controller/pembaca.js

```diff
@@ -480,29 +498,94 @@ exports.getUserNotification = async (req, res, next) => {
   }
 };
 
-/**
- * @author 16 MN
- *
- * Fungsi untuk membuat token pembaca
- */
-const createPembacaToken = async (pembaca) => {
+/*
+@author 16 MN
+membuat accestoken dari refresh token yang diberikan oleh pembaca
+*/
+exports.getAccessToken = async (req , res , next ) => {
+  try{
+    const refreshToken = req.signedCookies['refresh'];
+    if(!refreshToken){
+      const error = new Error('Not auhtorized');
+      error.statusCode = 403;
+      throw error;
+    }
+    const decodedRefreshToken = jwt.verify(refreshToken , process.env.JWT_REFRESH_KEY);
+    
+    if(decodedRefreshToken.roles !== process.env.PEMBACA_PREFIX){
+      const error = new Error('Forbidden access');
+      error.statusCode = 401;
+      error.cause = 'You are not allowed to access this route , please contact your administrator'
+      throw error;
+    }
+    const pembacaId = decodedRefreshToken.id
+    const pembaca = await Pembaca.findByPk(pembacaId);
+    
+    if(!pembaca){
+      const error = new Error('User with such token not found');
+      error.statusCode = 401;
+      error.cause = 'You are account might be deleted'
+      throw error;
+    }
+    
+    if(pembaca.last_changed_pwd > decodedRefreshToken.iat){
+      const error = new Error('Invalid token');
+      error.statusCode = 401;
+      error.cause = 'You might re-login to access this route'
+      throw error;      
+    }
+    
+    const newToken = await createAccessToken(pembaca);
+
+    res.status(201).json({
+      message : 'success create new access token',
+      token : newToken
+    })
+
+  }catch(err){
+    next(err);
+  }
+}
+
+/*
+@author 16 MN
+membuat refresh token bagi pembaca
+*/
+const createRefreshToken = async (pembaca) => {
+  const token = jwt.sign(
+    {
+      id: pembaca.id_pembaca,
+      roles: process.env.PEMBACA_PREFIX,
+    },
+    process.env.JWT_REFRESH_KEY,
+    { expiresIn: 60 * 60 * 24 * 7}
+  );
+  return token; 
+}
+
+/*
+@author 16 MN
+membuat access token bagi pembaca
+*/
+
+const createAccessToken = async (pembaca) => {
   const accessToken = jwt.sign(
     {
       id: pembaca.id_pembaca,
       roles: process.env.PEMBACA_PREFIX,
     },
     process.env.JWT_SECRET_KEY,
-    { expiresIn: 60 * 15 }
+    { expiresIn: 60 * 60 *  2  }
```

Comment:
ini expirenya berapa berarti? ..can you put some comment on calculations?

Reply:
done